### PR TITLE
Set retry policy

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Set retry policy for CAS operation
**Which issue(s) this PR fixes**:
Fixes #750 

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by maheshrajamani** [comment]
This is manually tested in local to confirm retry works on inserts, logs below.

INFO  [s0-io-1] 2024-01-11 14:24:41,983 CqlProxyRetryPolicy.java:69 - Retrying write timeout for request type : CAS  with retryCount: 1
INFO  [s0-io-1] 2024-01-11 14:24:42,495 CqlProxyRetryPolicy.java:69 - Retrying write timeout for request type : CAS  with retryCount: 2

File: src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlProxyRetryPolicy.java [hunk 1]
```diff
@@ -20,10 +21,13 @@
  * the intermediate layer is unavailable for some reason. In all other cases, this will rethrow the
  * exception to avoid retrying the requests at the driver level.
  */
+@RegisterForReflection
 public class CqlProxyRetryPolicy implements RetryPolicy {
   private static final Logger LOG = LoggerFactory.getLogger(CqlProxyRetryPolicy.class);
   private final String logPrefix;
 
+  private static final int MAX_RETRIES = 3;
```
Comment (jeffreyscarpenter) [inline src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlProxyRetryPolicy.java:29]: should this be configurable?

Comment (jeffreyscarpenter) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlProxyRetryPolicy.java:29]: I think we need to be able to tune this. Assuming that one of the possible errors is a timeout, the timeout value is going to be settable in profile. We need to be able to tune the two properties together so we can have control over the max time we might spend (timeout X retries?).

Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlProxyRetryPolicy.java:29]: There is no driver configuration to set maximum retries. Set to 3 based on Aaron input. One option is to use System.getProperties().

Comment (jeffreyscarpenter) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlProxyRetryPolicy.java]: right, we'd have to add our own property

**Review by jeffreyscarpenter — APPROVED** [review]

**Pull request merged**